src/main/kotlin/org/rust/ide/refactoring/move/RsMoveFilesOrDirectoriesHandler.kt

```diff
@@ -77,19 +104,17 @@ class RsMoveFilesOrDirectoriesHandler : MoveFilesOrDirectoriesHandler() {
     }
 }
 
-private fun PsiElement.canBeMoved(): Boolean {
-    if (this !is RsFile) return false
-
-    // TODO: support move files with its child modules
-    if (ownsDirectory) return false
-    if (hasChildModules()) return false
-
-    // TODO: support path attribute on mod declaration
-    if (pathAttribute != null) return false
-
-    return modName != null
+private fun RsFile.canBeMoved(): Boolean =
+    modName != null
         && crateRoot != null
         && crateRelativePath != null
         && !isCrateRoot
         && isEdition2018  // TODO: support 2015 edition
-}
+        && pathAttribute == null  // TODO: support path attribute on mod declaration
+
+fun PsiElement.adjustForMove(): RsFile? =
```

Comment:
It is not clear that **public** method `PsiElement.adjustForMove()` is only for files moving (but not just `PsiElement`s moving). Maybe we can rename this method or put it inside a companion object of some class related to files moving only (like `RsMoveFilesOrDirectoriesHandler`)?